src/main/kotlin/objects/BoardStateManager.kt

```diff
@@ -89,15 +89,128 @@ class BoardStateManager(
         this.boardState.ram[playerToIndex(player)] = getMaxRam(player)
     }
 
-    suspend fun handleSummonPacket(packet: SummonRequestPacket, player: Player) {
+    // TODO: remove this function after deck masters are no longer null
+    // This is here to encapsulate code that doesn't need to be in the
+    // final server
+    private suspend fun temporarySpecialLogicWithGameOverHandler(
+        deckMasterPlayer1: CardState?,
+        deckMasterPlayer2: CardState?,
+    ): Boolean {
+        val player1Died = (deckMasterPlayer1?.run { health < 0 }) ?: false
+        val player2Died = (deckMasterPlayer2?.run { health < 0 }) ?: false
+        if (player1Died && player2Died) {
+            return false
+        }
+
+        if (player1Died) {
+            getConnection(Player.Player1).sendPacket(GameOverPacket(false))
+            getConnection(Player.Player2).sendPacket(GameOverPacket(true))
+            return true
+        }
+
+        if (player2Died) {
+            getConnection(Player.Player1).sendPacket(GameOverPacket(true))
+            getConnection(Player.Player2).sendPacket(GameOverPacket(false))
+            return true
+        }
+
+        return false
+    }
+
+    suspend fun withGameOverHandler(handler: suspend () -> Unit) {
```

Comment:
Yep, it can. This is designed to easily add game over functionality to only select handlers if we want to, but if we're ok with calculating game over after every handler, this can just be a simple function.

Reply:
This is just the difference of 
``` kotlin
withGameOverHandler {
    handleXYZPacket()
}
```
vs 
``` kotlin
handleXYZPacket()
gameOverHandler()
```
at the caller. It doesn't matter for one, but if we have many whenXYZHandler's at some point it'll probably become annoying having all of them nested. Unless I'm missing a benefit of the first form I think the second one is simpler and easier to work with.